homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include <algorithm>
+#include <iostream>
+
+#include "compression.hpp"
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap) {
+    CompressedBitmap compressed{};
+    compressed.reserve(width * height);
+
+    std::for_each(bitmap.begin(), bitmap.end(), [&](auto& array) {
```

Comment:
for_each for me is similar to for. You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )